src/app/util/Request/Request.js

```diff
@@ -39,7 +50,7 @@ const appendTokenToHeaders = (headers) => {
 const formatURI = (query, variables, url) => {
     const stringifyVariables = Object.keys(variables).reduce(
         (acc, variable) => [...acc, `${ variable }=${ variables[ variable ] }`],
-        [`?hash=${ hash(query) }`]
+        [`?hash=${ hash(query + currencyCode) }`]
```

Comment:
I do not think this is okey. @eli-l Please advice - can not we just whitelist header to be used for caching? Or it URL only?